### ISSUE TITLE
Not works for *.kt - kotlin files

### ISSUE THREAD

**Issue by tomtom-gangdam-style-for-map** [hierynomus/license-gradle-plugin#155]
com.android.tools.build:gradle:3.1.0-beta2 
gradle wrapper 4.4 
Working for java - but for *.kt not even when I try: 

license {
    header rootProject.file('LICENSE')
    mapping("kt", "SLASHSTAR_STYLE")
    includes(["**/*.kt", "**/*.java"])
    strictCheck true
    excludes(["**/*Tests.java", "**/*.xml", "**/*TestsUtils.java", "**/*.json", "com/android/**/*.java"])
}

**Comment by widarlein** [comment]
Yeah, I have this problem as well!

**Comment by ysb33r** [comment]
This is a general problem. When the extension is not recognised, it does not even help to create a mapping.

It would be nice to have the ability to add custom extensions, not just language names to a mapping.

**Comment by hierynomus** [comment]
What is the complete `build.gradle` file? It should definitely be possible to add new mappings.

**Comment by ysb33r** [comment]
Here's is one example: https://gitlab.com/ysb33rOrg/java-nio2-providers/blob/development/build.gradle#L221

**Comment by widarlein** [comment]
@hierynomus my build.gradle isn't available publicly yet, so here it is:
```
plugins {
    id "com.github.hierynomus.license" version "0.14.0"
}
apply plugin: 'com.android.application'
apply plugin: 'kotlin-android'
apply plugin: 'kotlin-android-extensions'
apply plugin: 'com.github.hierynomus.license'

android {
    compileSdkVersion 25
    buildToolsVersion "26.0.2"

    defaultConfig {
        applicationId "se.geecity.android.steerandput"
        minSdkVersion 14
        targetSdkVersion 23
        manifestPlaceholders = [mapsApiKey: MAPS_API_KEY]
    }

    buildTypes {

        debug {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt') //, 'proguard-rules.txt'
        }

        release {
            minifyEnabled true
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.txt'
        }

        buildTypes.each {
            it.buildConfigField 'String', 'BICYCLESERVICE_API_KEY', '"$BICYCLESERVICE_API_KEY"'
        }
    }
    sourceSets {
        main.java.srcDirs += 'src/main/kotlin'
    }
}

license {
    header rootProject.file('mit.header')
    exclude('**/AutoValue_*.java')
    strictCheck true
    mapping('java', 'SLASHSTAR_STYLE')
    mapping('kt', 'SLASHSTAR_STYLE')
    ext.year = '2018'
    ext.name = 'Alexander Widar'
}

dependencies {
    compile 'com.google.code.gson:gson:2.8.0'
    compile 'com.android.support:support-v4:25.4.0'
    compile 'com.android.support:appcompat-v7:25.4.0'
    compile 'com.android.support:design:25.4.0'
    compile 'com.android.support:cardview-v7:25.4.0'
    compile 'com.google.android.gms:play-services-location:11.0.1'
    compile 'com.google.android.gms:play-services-maps:11.0.1'
    compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
    compile 'com.squareup.okhttp3:okhttp:3.7.0'
    compile 'com.github.PhilJay:MPAndroidChart:v3.0.2'
    compile 'org.koin:koin-android:0.8.0'
}
repositories {
    mavenCentral()
    maven {
        url 'https://maven.google.com/'
        name 'Google'
    }
}
```

**Comment by yogitas** [comment]
I could add this for kotlins file with by just adding extra task like this
task licenseFormatKotlin(type: com.hierynomus.gradle.license.tasks.LicenseFormat) {
    source = fileTree(dir: "src").include("**/*.kt")
}
licenseFormat.dependsOn licenseFormatKotlin

**Comment by yogitas** [comment]
it doesnot recognize kotlin as java project. Henec you have to add 
task licenseFormatForKotlinl(type: com.hierynomus.gradle.license.tasks.LicenseFormat) {
    source = fileTree(dir: <kotlin project directory>).include("**/*.kt")
}
licenseFormat.dependsOn licenseFormatForKotlinl

now if you run ./gradlew licenseFormat  it should consider kotlin files too

**Comment by utybo** [comment]
It seems to work without any issue now for .kt files, this should probably be closed

**Comment by widarlein** [comment]
It only works for me when adding @yogitas snippet. I have my kotlin code in the regular java source set, so I changed `<kotlin project directory>` to `src/main/java`

**Comment by utybo** [comment]
> I have my kotlin code in the regular java source set

That's a strange organization... It worked without any issue for me by placing the Kotlin code in the correct folder `src/main/kotlin`

**Comment by widarlein** [comment]
> That's a strange organization...

Not at all, the kotlin language reference states that _In mixed-language projects, Kotlin source files should reside in the same source root as the Java source files, and follow the same directory structure _ https://kotlinlang.org/docs/reference/coding-conventions.html#directory-structure

Also android developer reference states that _By default, new Kotlin files are saved into src/main/java/. You might find it's easier to see both Kotlin and Java files in one location. But if you'd prefer to separate your Kotlin files from your Java files, you can put Kotlin files under src/main/kotlin/ instead._

There's nothing inherently "correct" about putting code in the `src/main/kotlin` folder

**Comment by utybo** [comment]
Huh, TIL, I didn't know that. Sorry about my assumptions!

**Comment by ysb33r** [comment]
> There's nothing inherently "correct" about putting code in the src/main/kotlin folder

FYI There is the matter of Gradle build cache optimisation w.r.t. to polyglot builds. The Gradle folks recommend to have different language types in separate folders. (I still mix Groovy & Java in the same folder as I am willing to sacrifice a couple of seconds of build time).

**Comment by gladed** [comment]
0.15.0 not working for *.kt files and @utybo's fix works for the `licenseFormat` task but unfortunately not the license check itself.

**Comment by bric3** [comment]
@gladed That's easy just duplicate the logic with the `LicenceCheck` task

```
tasks.register("licenseCheckForKotlin", com.hierynomus.gradle.license.tasks.LicenseCheck::class) {
    source = fileTree(project.projectDir) { include("**/*.kt") }
}
tasks["license"].dependsOn("licenseCheckForKotlin")
tasks.register("licenseFormatForKotlin", com.hierynomus.gradle.license.tasks.LicenseFormat::class) {
    source = fileTree(project.projectDir) { include("**/*.kt") }
}
tasks["licenseFormat"].dependsOn("licenseFormatForKotlin")
```

Although unrecognized extensions are not handled, eg if the above snippet includes `"**/*.kts"` as well the plugin reports `Unknown file extension`

**Comment by leinardi** [comment]
What's the current status on Kotlin support?
